src/gui/folderman.cpp

```diff
@@ -1203,37 +1202,50 @@ QString FolderMan::checkPathValidityForNewFolder(const QString& path, bool forNe
         }
         if( ! folderDir.endsWith(QLatin1Char('/')) ) folderDir.append(QLatin1Char('/'));
 
-        if (QDir::cleanPath(f->path()) == QDir::cleanPath(userInput)
-                && QDir::cleanPath(QDir(f->path()).canonicalPath()) == QDir(userInput).canonicalPath()) {
```

Comment:
I suppose there was an error in the old code.  the `&&` should have been a `||` 


Anyway, this shows a flaw in the code that we should check for cannonical path everywhere, really.
Also we should use Case insensitive comparison depending on the file system.